ethportal-api/src/types/network.rs

```diff
@@ -29,32 +33,109 @@ impl std::str::FromStr for Network {
 }
 
 /// Enum for various different portal subnetworks in a "core" network.
-#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
 pub enum Subnetwork {
     Beacon,
     History,
     State,
+    CanonicalIndices,
+    VerkleState,
+    TransactionGossip,
+    Utp,
 }
 
 impl fmt::Display for Subnetwork {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
         match self {
-            Subnetwork::Beacon => write!(f, "beacon"),
-            Subnetwork::History => write!(f, "history"),
-            Subnetwork::State => write!(f, "state"),
+            Subnetwork::Beacon => write!(f, "Beacon"),
+            Subnetwork::History => write!(f, "History"),
+            Subnetwork::State => write!(f, "State"),
+            Subnetwork::CanonicalIndices => write!(f, "Canonical Indices"),
+            Subnetwork::VerkleState => write!(f, "Verkle State"),
+            Subnetwork::TransactionGossip => write!(f, "Transaction Gossip"),
+            Subnetwork::Utp => write!(f, "uTP"),
         }
     }
 }
 
-impl std::str::FromStr for Subnetwork {
-    type Err = String;
-
-    fn from_str(s: &str) -> Result<Self, Self::Err> {
+impl Subnetwork {
+    pub fn from_cli_arg(s: &str) -> Result<Self, String> {
         match s {
             "beacon" => Ok(Subnetwork::Beacon),
             "history" => Ok(Subnetwork::History),
             "state" => Ok(Subnetwork::State),
+            "canonical_indices" => Ok(Subnetwork::CanonicalIndices),
+            "verkle_state" => Ok(Subnetwork::VerkleState),
+            "transaction_gossip" => Ok(Subnetwork::TransactionGossip),
+            "utp" => Ok(Subnetwork::Utp),
             _ => Err(format!("Unknown subnetwork: {s}")),
         }
     }
+
+    pub fn to_cli_arg(&self) -> String {
+        match self {
+            Subnetwork::Beacon => "beacon".to_string(),
+            Subnetwork::History => "history".to_string(),
+            Subnetwork::State => "state".to_string(),
+            Subnetwork::CanonicalIndices => "canonical_indices".to_string(),
+            Subnetwork::VerkleState => "verkle_state".to_string(),
+            Subnetwork::TransactionGossip => "transaction_gossip".to_string(),
+            Subnetwork::Utp => "utp".to_string(),
+        }
+    }
+}
+
+impl From<Subnetwork> for u8 {
+    fn from(subnetwork: Subnetwork) -> u8 {
+        match subnetwork {
+            Subnetwork::Beacon => 1,
+            Subnetwork::History => 0,
+            Subnetwork::State => 2,
+            Subnetwork::CanonicalIndices => 4,
+            Subnetwork::VerkleState => 5,
+            Subnetwork::TransactionGossip => 3,
+            Subnetwork::Utp => 99,
+        }
+    }
+}
```

Comment:
To my knowledge this conversion `impl From<Subnetwork> for u8 {` is only used for the database

Reply:
Maybe Milos db refactor removed the need for `impl From<Subnetwork> for u8 {`, probably something we should double check nontheless

Reply:
I can't find anywhere we actually use it, so I've just removed it

Reply:
Sounds good, I assume it's usage has been removed then

Reply:
One thing to note is Enum's are just `variable` to `number` links. Where the u8 value would be the order of the variables in the Enum

```rust
#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
pub enum Subnetwork {
    Beacon = 0,
    History = 1,
    State = 2,
    CanonicalIndices = 3,
    VerkleState = 4,
    TransactionGossip = 5,
    Utp = 6,
}

```

So to my knowledge this would be implicit ^

```rust
#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
pub enum Subnetwork {
    Beacon,
    History,
    State,
    CanonicalIndices,
    VerkleState
    TransactionGossip,
    Utp,
}

```

Both of versions are equivalent, one just has implicit indexing and the other it is declared

Reply:
Yeah I think we are good